charts/matrix-stack/templates/ess-library/_ingress.tpl

```diff
@@ -35,25 +35,28 @@ annotations:
 {{- with required "element-io.ess-library.ingress.tlsSecret missing context" .context -}}
 {{- $ingress := required "element-io.ess-library.ingress.tls missing ingress" .ingress -}}
 {{- $host := .host | default $ingress.host -}}
+{{- $tlsEnabled := and $root.Values.ingress.tlsEnabled .ingress.tlsEnabled -}}
 {{- $ingressName := required "element-io.ess-library.ingress.tls missing ingressName" .ingressName -}}
-{{- with (include "element-io.ess-library.ingress.tlsSecret" (dict "root" $root "context" (dict "hosts" (list $host) "tlsSecret" $ingress.tlsSecret "ingressName" $ingressName))) }}
+{{- if $tlsEnabled }}
 tls:
+{{- with (include "element-io.ess-library.ingress.tlsSecret" (dict "root" $root "context" (dict "hosts" (list $host) "tlsSecret" $ingress.tlsSecret "ingressName" $ingressName))) }}
 {{ . | nindent 2 }}
 {{- end -}}
 {{- end -}}
 {{- end -}}
+{{- end -}}
 
 {{- define "element-io.ess-library.ingress.tlsSecret" -}}
 {{- $root := .root -}}
 {{- with required "element-io.ess-library.ingress.tlsSecret missing context" .context -}}
 {{- $ingressName := required "element-io.ess-library.ingress.tlsSecret missing ingress name" .ingressName -}}
```

Comment:
```suggestion
{{- with (include "element-io.ess-library.ingress.tlsHostsSecret" (dict "root" $root "context" (dict "hosts" (list $host) "tlsSecret" $ingress.tlsSecret "ingressName" $ingressName))) }}
{{ . | nindent 2 }}
{{- end -}}
{{- end -}}
{{- end -}}
{{- end -}}

{{- define "element-io.ess-library.ingress.tlsHostsSecret" -}}
{{- $root := .root -}}
{{- with required "element-io.ess-library.ingress.tlsHostsSecret missing context" .context -}}
{{- $ingressName := required "element-io.ess-library.ingress.tlsHostsSecret missing ingress name" .ingressName -}}
```

Doesn't make sense to call this `tlsSecret` if the secret part may not be there at all